[feat] Remove unnecessary Slurm polling when retrieving job pending reason (#3642)
eth-cscs/reframe: docs/config_reference.rst

   Ignore the ``ReqNodeNotAvail`` Slurm state.

   If a job associated to a test is in pending state with the Slurm reason ``ReqNodeNotAvail``, ReFrame will cancel the job.

>    Sometimes, however, when Slurm's backfill algorithm takes too long to compute, Slurm will set the pending reason to ``ReqNodeNotAvail`` and mark all system nodes as unavailable, causing ReFrame to its jobs.
>    In such cases, you may set this parameter to :obj:`True` to avoid this.

Reviewer: Typo "causing ReFrame to ___ its jobs"

```suggestion
   Sometimes Slurm’s backfill scheduling can take so long to compute that it temporarily sets the pending reason to ``ReqNodeNotAvail`` and reports all nodes as unavailable. ReFrame may interpret this condition and cancel its jobs. If you encounter this scenario, set this parameter to :obj:`True` to prevent ReFrame from killing jobs with the ``ReqNodeNotAvail`` reason.
```

Author: I applied these change exactly, except that I put every sentence in a separate line.

   .. versionadded:: 4.8

.. py:attribute:: systems.partitions.sched_options.slurm_job_cancel_reasons

   :required: No
   :default: ``["ReqNodeNotAvail"]``

   Reasons to proactively cancel a pending Slurm job.

>    If a job associated to a test is in pending state with one of the reasons listed here, ReFrame will cancel the job.

Reviewer: ```suggestion
   If a job associated to a test is in pending state with one of the reasons listed, ReFrame will cancel the job.
```

>    This option is relevant for the Slurm backends only.

Reviewer: ```suggestion
   This option is relevant for the Slurm backend only.
```

Author: I would keep the plural here, because there are two Slurm backends: `slurm` and `squeue`.
